packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -634,6 +635,13 @@ function allocateInSlot(vm: VM, children: VNodes) {
             continue;
         }
 
+        // Dive one level further if the content is wrapped in a VFragment
+        if (isVFragment(vnode)) {
+            // Since the VFragment children are delimited by empty text nodes, we can ignore those
+            allocateInSlot(vm, vnode.children.slice(1, -1));
```

Comment:
Since we only really look at the immediate child nodes, I figured the main issue with VFragment nodes is that we're not properly bypassing those nodes to look at the children in the fragment.

What this change does is add a bit of traversal to slot allocation, so we now traverse down to the children of the VFragment nodes we find.

This does mean that if we were to have something like
```
VFragment: [
    VFragment: [
        VFragment: [
            SomeOtherNode: ...
        ]
    ]
]
```
this function would continue traversing down the tree until it hits `SomeOtherNode`. I think this is valid though, because it would potentially map to this use case:
```
<template>
    <x-child-with-named-slots>
        <template lwc:if={condition1}>
            <template lwc:if={condition2}>
                <template lwc:if={condition3}>
                    <div slot="slotname1">Named slot content from parent</div>
                </template>
                ...
            </template>
            ...
        </template>
    </x-child-with-named-slots>
</template>
```
which, although could be simplified with JS, is technically a valid usecase especially if there's other content being branched off.

Reply:
Note: I removed the splicing so that the synthetic shadow behavior matches the native shadow behavior. Originally I was hoping we could ignore the text nodes, but native includes them. Open to other suggestions or thoughts on whether we should and can hide the text nodes here in native:

**Before**:
no name - synthetic: `assignedNodes.length === 1`
no name - native: `assignedNodes.length === 3` (includes the text nodes)
named - synthetic: `assignedNodes.length === 1`
named - native: `assignedNodes.length === 1`

**Now**:
no name - synthetic: `assignedNodes.length === 3` (includes the text nodes)
no name - native: `assignedNodes.length === 3` (includes the text nodes)
named - synthetic: `assignedNodes.length === 1`
named - native: `assignedNodes.length === 1`